### DISCUSSION TITLE
How do I properly setup LSP for Cpp and fix missing items

### DISCUSSION THREAD

**Discussion by OpenFoam-User** [helix-editor/helix#4432] [Q&A] [answered]
Hello,

When I run: `hx --health cpp`, I get the following output:

```
hx --health cpp
Configured language server: clangd
Binary for language server: Not found in $PATH
Configured debug adapter: lldb-vscode
Binary for debug adapter: Not found in $PATH
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```

How can I fix all the items? I mean for example how can I fix the "Binary for language server" and make it detectable in the PATH? and what is the "binary for language server" in the first place?

Thanks

**Comment by the-mikedavis** [comment, marked as answer]
Language Servers are standalone executables - for C/C++ it's [`clangd`](https://clangd.llvm.org/), for Rust it's [`rust-analyzer`](https://github.com/rust-lang/rust-analyzer). In order to get Language Server features, you'll need to install the Language Server and make sure the binary is available in `$PATH`: you should be able to invoke it on the command line.

You can install Language Servers through whatever package manager is used on your distribution (apt, brew, chocolatey, etc.) or follow the installation instructions directly for the Language Server. https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers has some notes and links on installing the default Language Servers.
